### PR TITLE
[Backport] [2.x] Restore support for Java 8 (#767)

### PR DESCRIPTION
**Pull request by reta**
Backport of https://github.com/opensearch-project/opensearch-java/pull/767 to `2.x`

### CODE REVIEW

File: java-client/build.gradle.kts [hunk 1]
```diff
@@ -146,15 +146,20 @@ val integrationTest = task<Test>("integrationTest") {
             System.getProperty("tests.awsSdk2support.domainRegion", "us-east-1"))
 }
 
+val opensearchVersion = "2.12.0-SNAPSHOT"
```
Comment (reta) [inline java-client/build.gradle.kts:149]: @dblock @VachaShah we may not be able to release the new client (with JDK 8 support) till OpenSearch 2.12.0 release, where the RestClient is also JDK-8 compatible

Comment (dblock) [reply java-client/build.gradle.kts:149]: Makes sense.

**Review by dblock — APPROVED** [review]
Any reason not to merge this until 2.12 is released?

**Comment by reta** [comment]
> Any reason not to merge this until 2.12 is released?

Thanks @dblock, only one reason - we could not make 2.x releases till 2.12.0 (since we cannot make a release with SNAPSHOT dependencies).

**Comment by dblock** [comment]
> > Any reason not to merge this until 2.12 is released?
> 
> Thanks @dblock, only one reason - we could not make 2.x releases till 2.12.0 (since we cannot make a release with SNAPSHOT dependencies).

Let's hold until then? We may need a security patch or an earlier release with other features.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@reta 🚢 ? Let's make a release?

**Comment by reta** [comment]
> @reta 🚢 ? Let's make a release?

@dblock I think we agreed to hold it back till 2.12.0 (as per https://github.com/opensearch-project/opensearch-java/pull/777#issuecomment-1866523642 and https://github.com/opensearch-project/opensearch-java/pull/777#issuecomment-1866508165)?

**Comment by dblock** [comment]
> > @reta 🚢 ? Let's make a release?
> 
> @dblock I think we agreed to hold it back till 2.12.0 (as per [#777 (comment)](https://github.com/opensearch-project/opensearch-java/pull/777#issuecomment-1866523642) and [#777 (comment)](https://github.com/opensearch-project/opensearch-java/pull/777#issuecomment-1866508165))?

Ah right! I already forgot what happened **last year**.

**Pull request merged**